Source/SwiftLintFramework/Rules/NimbleOperatorRule.swift

```diff
@@ -39,25 +40,118 @@ public struct NimbleOperatorRule: ConfigurationProviderRule, OptInRule {
+                location: Location(file: file, characterOffset: $0.location))
         }
     }
+
+    private func violationMatchesRanges(in file: File) -> [NSRange] {
+        let operatorNames = operatorsMapping.keys
+        let operatorsPattern = "(" + operatorNames.joined(separator: "|") + ")"
+
+        let variablePattern = "(.(?!expect\\())+?"
+        let pattern = "expect\\(\(variablePattern)\\)\\.to(Not)?\\(\(operatorsPattern)\\(\(variablePattern)\\)\\)"
+
+        let excludingKinds = SyntaxKind.commentKinds()
+
+        return file.matchPattern(pattern)
+            .filter { _, kinds in
+                kinds.filter(excludingKinds.contains).isEmpty && kinds.first == .identifier
+            }.map { $0.0 }
+    }
+
+    public func correctFile(_ file: File) -> [Correction] {
+        let matches = violationMatchesRanges(in: file)
+            .filter { file.ruleEnabledViolatingRanges([$0], forRule: self).isEmpty == false }
+        guard matches.isEmpty == false else { return [] }
+
+        let description = type(of: self).description
+        var corrections: [Correction] = []
+        var contents = file.contents
+
+        matchesLoop: for range in matches.sorted(by: { $0.location > $1.location }) {
+            for (functionName, operatorCorrections) in operatorsMapping {
+                guard let correctedString = contents.replace(function: functionName,
+                                                             with: operatorCorrections,
+                                                             in: range)
+                else {
+                    continue
+                }
+
+                contents = correctedString
+                corrections.append(Correction(ruleDescription: description,
+                                              location:  Location(file: file, characterOffset: range.location)))
+                break matchesLoop
+            }
+        }
+
+        file.write(contents)
+        return corrections
+    }
+}
+
+extension String {
+    /// Returns corrected string if the correction is possible, otherwise returns nil.
+    fileprivate func replace(function name: NimbleOperatorRule.MatcherFunction,
+                             with operators: NimbleOperatorRule.Operators,
+                             in range: NSRange) -> String? {
+
+        let anything = "\\s*(.*?)\\s*"
+
+        let toPattern = ("expect\\(\(anything)\\)\\.to\\(\(name)\\(\(anything)\\)\\)", operators.to)
+        let toNotPattern = ("expect\\(\(anything)\\)\\.toNot\\(\(name)\\(\(anything)\\)\\)", operators.toNot)
+
+        var correctedString: String? = nil
+
+        for (pattern, operatorString) in [toPattern, toNotPattern] {
+            guard let operatorString = operatorString else {
+                continue
+            }
+
+            let expression = regex(pattern)
+            if let _ = expression.matches(in: self, options: [], range: range).first {
```

Comment:
Please use `if ... != nil` rather than `if let _ = ...`. We ought to have a rule for this.

And while we're at it, instead of getting the first item only to discard it, might as well check `isEmpty`:

```swift
if !expression.matches(in: self, options: [], range: range).isEmpty {
```